### PR TITLE
[sBTC DR] Add check for contract versions via clarinet-sdk

### LINKED ISSUE #172: [sBTC DR] Introduce clarinet-sdk to this repo
Two days ago, clarinet-sdk v0.4.0 (https://www.npmjs.com/package/@hirosystems/clarinet-sdk) was published that has enough features to replace the current setup for clarity testing.

Using clarinet-sdk means moving from deno to node.js. Then, testing is done using vitest

### PR DESCRIPTION
**Pull request by friedger**
## Description
This PR 
* adds a unit test to check that the contract versions of `clarity-bitcoin-mini` differ only in DEBUG flag
* introduces clarinet-sdk to this repo
* fixes #172

### CODE REVIEW

**Comment by stjepangolemac** [comment]
What does this concretely mean for Romeo? Does it change how we deploy or test contracts? Is it related to interpolating env vars into contracts that we mentioned the other day?

**Comment by friedger** [comment]
If adopted, Clarinet-sdk changes the way we test contracts. 

Clarinet-sdk can create new contracts based on env vars.

For deployments, I am not sure whether the SDK supports that or we continue with the current solution. Maybe @hugocaillard can comment.

**Comment by hugocaillard** [comment]
The clarinet-sdk only replaces the `clarinet test` and `clarinet run` features. It has no impact on deployment.

This PR looks good to me 👌 

You may have one issue **while moving from `clarinet test` to the SDK** for testing, regarding the test file notation with `.test.ts` or `_test.ts`:

- `npm test` (with vitest) won't try to run the `tests/*_test.ts` files (by default, vitest checks `*.test.ts`)
- **but** i think that `clarinet test` might try to execute both `*_test.ts` and `*.test.ts`


I think you'll need to run `clarinet test tests/*_test.ts` instead of just `clarinet test`. And again, this is just until all tests are migrated to Node.js

**Comment by lgalabru** [comment]
I think `clarinet-sdk` would simplify a lot of things in this repo, for instance, the whole layer generating tests [here](https://github.com/stacks-network/sbtc/blob/main/romeo/asset-contract/ext/generate-tests.ts) could be dropped.
This new sdk was designed to:
- help developers test their smart contracts (so similar to `clarinet test`)
- help developers design their own test harnessing framework, by exposing all the functions for building on top Clarity VMs at a low level, which is new. 
With this new design, creating a npm library that would help developers writing clarity for testing clarity, or a lib for fuzz testing clarity code should be relatively straightforward and more robust.
Feel free to ping us if you think that a meeting / crash course could help and accelerate this transition.

File: romeo/asset-contract/tests/clarity-bitcoin-mini-deploy.test.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { beforeEach, describe, expect, it } from "vitest";
+
+beforeEach(async () => {
+  await vm.initSession(process.cwd(), "./Clarinet.toml");
+});
+
+describe("test deploy version and unit test version of clarity-bitcoin-mini", () => {
+  it("ensures the versions differ only in DEBUG flag", () => {
```
Comment (setzeus) [inline romeo/asset-contract/tests/clarity-bitcoin-mini-deploy.test.ts:8]: Appreciate the comment here - answers my question on what's the difference b/w both contract.

**Review by setzeus — APPROVED** [review]
Changes LGTM but can you point me to more context / docs on Clarinet-SDK?

**Comment by hugocaillard** [comment]
@setzeus There's some context on [discord](https://discord.com/channels/621759717756370964/839633619261456444/1154096301986811975). But the official docs and a blog post will be available this week, along with the v1.0

File: romeo/asset-contract/tests/clarity-bitcoin-mini-deploy.test.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { beforeEach, describe, expect, it } from "vitest";
+
+beforeEach(async () => {
+  await vm.initSession(process.cwd(), "./Clarinet.toml");
+});
```
Comment (hugocaillard) [inline romeo/asset-contract/tests/clarity-bitcoin-mini-deploy.test.ts:5]:
```suggestion
```

This is now the default behavior of the test framework, so it's not needed anymore.

And this it what appears to make the test fail right now.

cc @friedger

File: romeo/asset-contract/package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "license": "MIT",
+  "type": "module",
+  "scripts": {
+    "generate:docs": "npx tsx ./scripts/generate-docs.ts",
+    "test": "vitest run",
+    "test:coverage": "vitest run -- --coverage true",
+    "test:watch": "chokidar \"unit-tests/**/*.ts\" \"contracts/**/*.clar\" -c \"npm t\""
+  },
+  "dependencies": {
+    "@hirosystems/clarinet-sdk": "^0.6.4",
```
Comment (hugocaillard) [inline romeo/asset-contract/package.json:15]:
```suggestion
    "@hirosystems/clarinet-sdk": "^0.9.0",
```

(will require to re-run `npm install` to apply changes in the package-lock.json)

File: romeo/asset-contract/package.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    "@stacks/transactions": "^6.8.1",
+    "@types/typescript": "^2.0.0",
+    "typescript": "^5.2.2",
+    "vite": "^4.4.9",
+    "vitest": "^0.34.4",
+    "vitest-environment-clarinet": "^0.1.1"
```
Comment (hugocaillard) [inline romeo/asset-contract/package.json:21]:
```suggestion
    "vitest-environment-clarinet": "^0.2.0"
```

(will require to re-run `npm install` to apply changes in the package-lock.json)

File: romeo/asset-contract/scripts/generate-docs.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import {
+  initVM,
```
Comment (hugocaillard) [inline romeo/asset-contract/scripts/generate-docs.ts:2]:
```suggestion
  initSimnet,
```

File: romeo/asset-contract/tests/clarity-bitcoin-mini-deploy.test.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { beforeEach, describe, expect, it } from "vitest";
+
+
+describe("test deploy version and unit test version of clarity-bitcoin-mini", () => {
+  it("ensures the versions differ only in DEBUG flag", () => {
+    const sourceForUnitTests = vm.getContractSource("clarity-bitcoin-mini")!;
+    const sourceForDeploy = vm.getContractSource(
```
Comment (hugocaillard) [inline romeo/asset-contract/tests/clarity-bitcoin-mini-deploy.test.ts:7]:
```suggestion
    const sourceForUnitTests = simnet.getContractSource("clarity-bitcoin-mini")!;
    const sourceForDeploy = simnet.getContractSource(
```

File: romeo/asset-contract/scripts/generate-docs.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  ContractInterfaceFunctionAccess,
+} from "@hirosystems/clarinet-sdk";
+
+async function main() {
+  const vm = await initVM();
+  for (let [contractId, abi] of vm.getContractsInterfaces()) {
+    const ast = vm.getContractAST(contractId);
+    const source = vm.getContractSource(contractId);
```
Comment (hugocaillard) [inline romeo/asset-contract/scripts/generate-docs.ts:10]:
```suggestion
  const simnet = await initVM();
  for (let [contractId, abi] of simnet.getContractsInterfaces()) {
    const ast = simnet.getContractAST(contractId);
    const source = simnet.getContractSource(contractId);
```

**Review by hugocaillard — CHANGES_REQUESTED** [review]
@friedger 

We wanted to make a pretty important breaking change before the release of v1.
The name `vm` (`ClarityVM` and `initVM`) didn't make sense as it kind of refers to the Clarity VM, but what we are really interacting with in the SDK is the `simnet`.

So we reworded that. PR [here](https://github.com/hirosystems/clarinet/pull/1181) for full context.

Thanks for bearing with us in these early stage version of the SDK. 

This v0.9 is very likely to be promoted to v1 🙌 

PS: ofc feel free to commit my suggestion or make your own commit with everything in it

**Review by setzeus — APPROVED** [review]

File: romeo/asset-contract/tests/clarity-bitcoin-mini-deploy.test.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  it("ensures the versions differ only in DEBUG flag", () => {
+    const sourceForUnitTests = simnet.getContractSource("clarity-bitcoin-mini")!;
+    const sourceForDeploy = simnet.getContractSource(
+      "clarity-bitcoin-mini-deploy"
+    )!;
+    expect(sourceForDeploy).not.toEqual(sourceForUnitTests);
+    expect(sourceForDeploy).toEqual(
+      sourceForUnitTests.replace(
+        "(define-constant DEBUG-MODE true)",
+        "(define-constant DEBUG-MODE false)"
```
Comment (MarvinJanssen) [inline romeo/asset-contract/tests/clarity-bitcoin-mini-deploy.test.ts:14]: Some extra white space can kill this test. For now it is OK, perhaps in the future we can compare AST.

**Review by MarvinJanssen — APPROVED** [review]

**Pull request merged**